sonic_package_manager/service_creator/feature.py

```diff
@@ -27,51 +35,93 @@ def register(self,
                  manifest: Manifest,
                  state: str = 'disabled',
                  owner: str = 'local'):
+        """ Register feature in CONFIG DBs.
+
+        Args:
+            manifest: Feature's manifest.
+            state: Desired feature admin state.
+            owner: Owner of this feature (kube/local).
+        Returns:
+            None.
+        """
+
         name = manifest['service']['name']
-        for table in self._get_tables():
-            cfg_entries = self.get_default_feature_entries(state, owner)
-            non_cfg_entries = self.get_non_configurable_feature_entries(manifest)
+        db_connectors = self._sonic_db.get_connectors()
+        cfg_entries = self.get_default_feature_entries(state, owner)
+        non_cfg_entries = self.get_non_configurable_feature_entries(manifest)
 
-            exists, current_cfg = table.get(name)
+        for conn in db_connectors:
+            current_cfg = conn.get_entry(FEATURE, name)
 
             new_cfg = cfg_entries.copy()
             # Override configurable entries with CONFIG DB data.
-            new_cfg = {**new_cfg, **dict(current_cfg)}
+            new_cfg = {**new_cfg, **current_cfg}
             # Override CONFIG DB data with non configurable entries.
             new_cfg = {**new_cfg, **non_cfg_entries}
 
-            table.set(name, list(new_cfg.items()))
+            conn.set_entry(FEATURE, name, new_cfg)
 
     def deregister(self, name: str):
-        for table in self._get_tables():
-            table._del(name)
+        """ Deregister feature by name.
+
+        Args:
+            name: Name of the feature in CONFIG DB.
+        Returns:
+            None
+        """
+
+        db_connetors = self._sonic_db.get_connectors()
+        for conn in db_connetors:
+            conn.set_entry(FEATURE, name, None)
+
+    def update(self,
+               old_manifest: Manifest,
+               new_manifest: Manifest):
+        """ Migrate feature configuration. It can be that non-configurable
+        feature entries have to be updated. e.g: "has_timer" for example if
+        the new feature introduces a service timer or name of the service has
+        changed, but user configurable entries are not changed).
+
+        Args:
+            old_manifest: Old feature manifest.
+            new_manifest: New feature manifest.
+        Returns:
+            None
+        """
+
+        old_name = old_manifest['service']['name']
+        new_name = new_manifest['service']['name']
+        db_connectors = self._sonic_db.get_connectors()
+        non_cfg_entries = self.get_non_configurable_feature_entries(new_manifest)
+
+        for conn in db_connectors:
+            current_cfg = conn.get_entry(FEATURE, old_name)
+            conn.set_entry(FEATURE, old_name, None)
```

Comment:
Does this mean, feature "snmp" could become "xyz". In other words the new manifest completely renames it ?